### ISSUE TITLE
pip install fails

### ISSUE THREAD

**Issue by dlfivefifty** [fruzsinaagocs/oscode#5]
```
solver@solver-mbook ~ % pip install pyoscode
Collecting pyoscode
  Downloading pyoscode-1.0.0.tar.gz (70.2 MB)
     |████████████████████████████████| 70.2 MB 498 kB/s 
Collecting numpy
  Using cached numpy-1.19.4-cp37-cp37m-macosx_10_9_x86_64.whl (15.3 MB)
Building wheels for collected packages: pyoscode
  Building wheel for pyoscode (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /usr/local/opt/python/bin/python3.7 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/tt/c5pb_r2131d25c9f_b2z7jrc0000gp/T/pip-install-qiallowx/pyoscode_a2904fd97ae3468e899ba2642c231cd0/setup.py'"'"'; __file__='"'"'/private/var/folders/tt/c5pb_r2131d25c9f_b2z7jrc0000gp/T/pip-install-qiallowx/pyoscode_a2904fd97ae3468e899ba2642c231cd0/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/var/folders/tt/c5pb_r2131d25c9f_b2z7jrc0000gp/T/pip-wheel-jlqo7fx3
       cwd: /private/var/folders/tt/c5pb_r2131d25c9f_b2z7jrc0000gp/T/pip-install-qiallowx/pyoscode_a2904fd97ae3468e899ba2642c231cd0/
  Complete output (73 lines):
  WARNING: pip is being invoked by an old script wrapper. This will fail in a future version of pip.
  Please see https://github.com/pypa/pip/issues/5599 for advice on fixing the underlying issue.
  To avoid this problem you can invoke Python with '-m pip' instead of running pip directly.
  Requirement already satisfied: numpy in /usr/local/lib/python3.7/site-packages (1.19.4)
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-10.15-x86_64-3.7
  creating build/lib.macosx-10.15-x86_64-3.7/pyoscode
  copying pyoscode/__init__.py -> build/lib.macosx-10.15-x86_64-3.7/pyoscode
  running egg_info
  writing pyoscode.egg-info/PKG-INFO
  writing dependency_links to pyoscode.egg-info/dependency_links.txt
  writing requirements to pyoscode.egg-info/requires.txt
  writing top-level names to pyoscode.egg-info/top_level.txt
  reading manifest file 'pyoscode.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  writing manifest file 'pyoscode.egg-info/SOURCES.txt'
  copying pyoscode/__init__.pyc -> build/lib.macosx-10.15-x86_64-3.7/pyoscode
  copying pyoscode/_pyoscode.cpp -> build/lib.macosx-10.15-x86_64-3.7/pyoscode
  copying pyoscode/_pyoscode.hpp -> build/lib.macosx-10.15-x86_64-3.7/pyoscode
  copying pyoscode/_python.hpp -> build/lib.macosx-10.15-x86_64-3.7/pyoscode
  creating build/lib.macosx-10.15-x86_64-3.7/pyoscode/__pycache__
  copying pyoscode/__pycache__/__init__.cpython-36.pyc -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/__pycache__
  copying pyoscode/__pycache__/__init__.cpython-38.pyc -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/__pycache__
  creating build/lib.macosx-10.15-x86_64-3.7/pyoscode/docs
  copying pyoscode/docs/Makefile -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/docs
  creating build/lib.macosx-10.15-x86_64-3.7/pyoscode/docs/source
  copying pyoscode/docs/source/conf.py -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/docs/source
  copying pyoscode/docs/source/index.rst -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/docs/source
  copying pyoscode/docs/source/introduction.rst -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/docs/source
  copying pyoscode/docs/source/oscode.rst -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/docs/source
  copying pyoscode/docs/source/pyoscode.rst -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/docs/source
  creating build/lib.macosx-10.15-x86_64-3.7/pyoscode/images
  copying pyoscode/images/airy-example.png -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/images
  copying pyoscode/images/burst-example.png -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/images
  copying pyoscode/images/gamma.png -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/images
  copying pyoscode/images/omega.png -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/images
  copying pyoscode/images/oscillator.gif -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/images
  copying pyoscode/images/oscillator.png -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/images
  copying pyoscode/images/spectra.gif -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/images
  running build_ext
  building '_pyoscode' extension
  Warning: Can't read registry to find the necessary compiler setting
  Make sure that Python modules winreg, win32api or win32con are installed.
  C compiler: clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/include -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/Tk.framework/Versions/8.5/Headers
  
  creating build/temp.macosx-10.15-x86_64-3.7
  creating build/temp.macosx-10.15-x86_64-3.7/pyoscode
  compile options: '-Iinclude -Ipyoscode -I/usr/local/include -I/usr/local/opt/openssl@1.1/include -I/usr/local/opt/sqlite/include -I/usr/local/Cellar/python/3.7.6_1/Frameworks/Python.framework/Versions/3.7/include/python3.7m -c'
  extra options: '-std=c++11 -Wall'
  clang: pyoscode/_pyoscode.cpp
  In file included from pyoscode/_pyoscode.cpp:4:
  In file included from include/system.hpp:2:
  include/interpolator.hpp:74:20: warning: unused variable 'x1' [-Wunused-variable]
              double x1 = xstart + (i+1)*dx;
                     ^
  include/interpolator.hpp:75:20: warning: unused variable 'x0' [-Wunused-variable]
              double x0 = xstart + i*dx;
                     ^
  In file included from pyoscode/_pyoscode.cpp:5:
  include/solver.hpp:280:38: warning: unused variable 'dx_dense_rk' [-Wunused-variable]
      std::complex<double> x_dense_rk, dx_dense_rk;
                                       ^
  include/solver.hpp:280:26: warning: unused variable 'x_dense_rk' [-Wunused-variable]
      std::complex<double> x_dense_rk, dx_dense_rk;
                           ^
  pyoscode/_pyoscode.cpp:7:10: fatal error: 'numpy/arrayobject.h' file not found
  #include <numpy/arrayobject.h>
           ^~~~~~~~~~~~~~~~~~~~~
  4 warnings and 1 error generated.
  error: Command "clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/include -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/Tk.framework/Versions/8.5/Headers -Iinclude -Ipyoscode -I/usr/local/include -I/usr/local/opt/openssl@1.1/include -I/usr/local/opt/sqlite/include -I/usr/local/Cellar/python/3.7.6_1/Frameworks/Python.framework/Versions/3.7/include/python3.7m -c pyoscode/_pyoscode.cpp -o build/temp.macosx-10.15-x86_64-3.7/pyoscode/_pyoscode.o -std=c++11 -Wall" failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for pyoscode
  Running setup.py clean for pyoscode
Failed to build pyoscode
Installing collected packages: numpy, pyoscode
    Running setup.py install for pyoscode ... error
    ERROR: Command errored out with exit status 1:
     command: /usr/local/opt/python/bin/python3.7 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/tt/c5pb_r2131d25c9f_b2z7jrc0000gp/T/pip-install-qiallowx/pyoscode_a2904fd97ae3468e899ba2642c231cd0/setup.py'"'"'; __file__='"'"'/private/var/folders/tt/c5pb_r2131d25c9f_b2z7jrc0000gp/T/pip-install-qiallowx/pyoscode_a2904fd97ae3468e899ba2642c231cd0/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/tt/c5pb_r2131d25c9f_b2z7jrc0000gp/T/pip-record-kbjhui_b/install-record.txt --single-version-externally-managed --compile --install-headers /usr/local/include/python3.7m/pyoscode
         cwd: /private/var/folders/tt/c5pb_r2131d25c9f_b2z7jrc0000gp/T/pip-install-qiallowx/pyoscode_a2904fd97ae3468e899ba2642c231cd0/
    Complete output (73 lines):
    WARNING: pip is being invoked by an old script wrapper. This will fail in a future version of pip.
    Please see https://github.com/pypa/pip/issues/5599 for advice on fixing the underlying issue.
    To avoid this problem you can invoke Python with '-m pip' instead of running pip directly.
    Requirement already satisfied: numpy in /usr/local/lib/python3.7/site-packages (1.19.4)
    running install
    running build
    running build_py
    creating build
    creating build/lib.macosx-10.15-x86_64-3.7
    creating build/lib.macosx-10.15-x86_64-3.7/pyoscode
    copying pyoscode/__init__.py -> build/lib.macosx-10.15-x86_64-3.7/pyoscode
    running egg_info
    writing pyoscode.egg-info/PKG-INFO
    writing dependency_links to pyoscode.egg-info/dependency_links.txt
    writing requirements to pyoscode.egg-info/requires.txt
    writing top-level names to pyoscode.egg-info/top_level.txt
    reading manifest file 'pyoscode.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    writing manifest file 'pyoscode.egg-info/SOURCES.txt'
    copying pyoscode/__init__.pyc -> build/lib.macosx-10.15-x86_64-3.7/pyoscode
    copying pyoscode/_pyoscode.cpp -> build/lib.macosx-10.15-x86_64-3.7/pyoscode
    copying pyoscode/_pyoscode.hpp -> build/lib.macosx-10.15-x86_64-3.7/pyoscode
    copying pyoscode/_python.hpp -> build/lib.macosx-10.15-x86_64-3.7/pyoscode
    creating build/lib.macosx-10.15-x86_64-3.7/pyoscode/__pycache__
    copying pyoscode/__pycache__/__init__.cpython-36.pyc -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/__pycache__
    copying pyoscode/__pycache__/__init__.cpython-38.pyc -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/__pycache__
    creating build/lib.macosx-10.15-x86_64-3.7/pyoscode/docs
    copying pyoscode/docs/Makefile -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/docs
    creating build/lib.macosx-10.15-x86_64-3.7/pyoscode/docs/source
    copying pyoscode/docs/source/conf.py -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/docs/source
    copying pyoscode/docs/source/index.rst -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/docs/source
    copying pyoscode/docs/source/introduction.rst -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/docs/source
    copying pyoscode/docs/source/oscode.rst -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/docs/source
    copying pyoscode/docs/source/pyoscode.rst -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/docs/source
    creating build/lib.macosx-10.15-x86_64-3.7/pyoscode/images
    copying pyoscode/images/airy-example.png -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/images
    copying pyoscode/images/burst-example.png -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/images
    copying pyoscode/images/gamma.png -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/images
    copying pyoscode/images/omega.png -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/images
    copying pyoscode/images/oscillator.gif -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/images
    copying pyoscode/images/oscillator.png -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/images
    copying pyoscode/images/spectra.gif -> build/lib.macosx-10.15-x86_64-3.7/pyoscode/images
    running build_ext
    building '_pyoscode' extension
    Warning: Can't read registry to find the necessary compiler setting
    Make sure that Python modules winreg, win32api or win32con are installed.
    C compiler: clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/include -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/Tk.framework/Versions/8.5/Headers
    
    creating build/temp.macosx-10.15-x86_64-3.7
    creating build/temp.macosx-10.15-x86_64-3.7/pyoscode
    compile options: '-Iinclude -Ipyoscode -I/usr/local/include -I/usr/local/opt/openssl@1.1/include -I/usr/local/opt/sqlite/include -I/usr/local/Cellar/python/3.7.6_1/Frameworks/Python.framework/Versions/3.7/include/python3.7m -c'
    extra options: '-std=c++11 -Wall'
    clang: pyoscode/_pyoscode.cpp
    In file included from pyoscode/_pyoscode.cpp:4:
    In file included from include/system.hpp:2:
    include/interpolator.hpp:74:20: warning: unused variable 'x1' [-Wunused-variable]
                double x1 = xstart + (i+1)*dx;
                       ^
    include/interpolator.hpp:75:20: warning: unused variable 'x0' [-Wunused-variable]
                double x0 = xstart + i*dx;
                       ^
    In file included from pyoscode/_pyoscode.cpp:5:
    include/solver.hpp:280:38: warning: unused variable 'dx_dense_rk' [-Wunused-variable]
        std::complex<double> x_dense_rk, dx_dense_rk;
                                         ^
    include/solver.hpp:280:26: warning: unused variable 'x_dense_rk' [-Wunused-variable]
        std::complex<double> x_dense_rk, dx_dense_rk;
                             ^
    pyoscode/_pyoscode.cpp:7:10: fatal error: 'numpy/arrayobject.h' file not found
    #include <numpy/arrayobject.h>
             ^~~~~~~~~~~~~~~~~~~~~
    4 warnings and 1 error generated.
    error: Command "clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/include -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/Tk.framework/Versions/8.5/Headers -Iinclude -Ipyoscode -I/usr/local/include -I/usr/local/opt/openssl@1.1/include -I/usr/local/opt/sqlite/include -I/usr/local/Cellar/python/3.7.6_1/Frameworks/Python.framework/Versions/3.7/include/python3.7m -c pyoscode/_pyoscode.cpp -o build/temp.macosx-10.15-x86_64-3.7/pyoscode/_pyoscode.o -std=c++11 -Wall" failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /usr/local/opt/python/bin/python3.7 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/tt/c5pb_r2131d25c9f_b2z7jrc0000gp/T/pip-install-qiallowx/pyoscode_a2904fd97ae3468e899ba2642c231cd0/setup.py'"'"'; __file__='"'"'/private/var/folders/tt/c5pb_r2131d25c9f_b2z7jrc0000gp/T/pip-install-qiallowx/pyoscode_a2904fd97ae3468e899ba2642c231cd0/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/tt/c5pb_r2131d25c9f_b2z7jrc0000gp/T/pip-record-kbjhui_b/install-record.txt --single-version-externally-managed --compile --install-headers /usr/local/include/python3.7m/pyoscode Check the logs for full command output.
```

**Comment by fruzsinaagocs** [comment]
Hi there! Thank you for submitting this issue.

- It turns out this was a nasty, OS X/clang-specific bug. I reproduced it with Travis CI. 
- `setup(...)` in the `setup.py` was not passing the numpy include directories `np.get_include()` up.
- Fixed it by adding `np.get_include()` to `Extension(..., include_dirs=[np.get_include()])`.
- Released the fix on pyPI: [pyoscode 1.0.2](https://pypi.org/project/pyoscode/) and updated the `master` branch.
- Also added OS X 10.15 to the platforms Travis CI will be testing on :innocent:

**Comment by dlfivefifty** [comment]
Works, thanks!
